core/sql/sqlcomp/CmpSeabaseDDLcommentOn.cpp

```diff
@@ -0,0 +1,202 @@
+            enMDObjType = COM_BASE_TABLE_OBJECT;
+            break;
+    
+        case StmtDDLCommentOn::COMMENT_ON_TYPE_COLUMN:
+            enMDObjType = COM_BASE_TABLE_OBJECT;
+            break;
+
+        case StmtDDLCommentOn::COMMENT_ON_TYPE_INDEX:
+            enMDObjType = COM_INDEX_OBJECT;
+            break;
+
+        case StmtDDLCommentOn::COMMENT_ON_TYPE_SCHEMA:
+            enMDObjType = COM_PRIVATE_SCHEMA_OBJECT;
+            break;
+
+        case StmtDDLCommentOn::COMMENT_ON_TYPE_VIEW:
+            enMDObjType = COM_VIEW_OBJECT;
+            break;
+
+        case StmtDDLCommentOn::COMMENT_ON_TYPE_LIBRARY:
+            enMDObjType = COM_LIBRARY_OBJECT;
+            break;
+
+        case StmtDDLCommentOn::COMMENT_ON_TYPE_PROCEDURE:
+        case StmtDDLCommentOn::COMMENT_ON_TYPE_FUNCTION:
+            enMDObjType = COM_USER_DEFINED_ROUTINE_OBJECT;
+            break;
+
+        default:
+            break;
+
+    }
+
+  NAString catalogNamePart = objectName.getCatalogNamePartAsAnsiString();
+  NAString schemaNamePart = objectName.getSchemaNamePartAsAnsiString(TRUE);
+  NAString objNamePart = objectName.getObjectNamePartAsAnsiString(TRUE);
+
+  const NAString extObjName = objectName.getExternalName(TRUE);
+
+  ExeCliInterface cliInterface(STMTHEAP, NULL, NULL,
+                                       CmpCommon::context()->sqlSession()->getParentQid());
+  Int64 objUID = 0;
+  Int32 objectOwnerID = SUPER_USER;
+  Int32 schemaOwnerID = SUPER_USER;
+  Int64 objectFlags = 0;
+
+  // Verify that the requester has COMMENT privilege.
+  if (isAuthorizationEnabled() && !ComUser::isRootUserID())
+    {
+      NAString privMgrMDLoc;
+      CONCAT_CATSCH(privMgrMDLoc, getSystemCatalog(), SEABASE_PRIVMGR_SCHEMA);
+
+      PrivMgrComponentPrivileges componentPrivileges(std::string(privMgrMDLoc.data()), CmpCommon::diags());
+
+      if (!componentPrivileges.hasSQLPriv(ComUser::getCurrentUser(), SQLOperation::COMMENT, true))
+      {
+         *CmpCommon::diags() << DgSqlCode(-CAT_NOT_AUTHORIZED);
+         processReturn ();
+         return;
+      }
+    }
+
+  //get UID of object
+  objUID = getObjectInfo(&cliInterface,
+                              catalogNamePart.data(), schemaNamePart.data(), objNamePart.data(), 
+                              enMDObjType,
+                              objectOwnerID,
+                              schemaOwnerID,
+                              objectFlags);
+  if (objUID < 0 || objectOwnerID == 0 || schemaOwnerID == 0)
+    {
+      CmpCommon::diags()->clear();
+      *CmpCommon::diags() << DgSqlCode(-1389)
+                          << DgString0(extObjName);
+      processReturn();
+      return;
+    }
+
+  // add, remove, change comment of object/column
+  const NAString & comment = commentOnNode->getComment();
+  char * query = new(STMTHEAP) char[comment.length()+1024];
+
+  if (StmtDDLCommentOn::COMMENT_ON_TYPE_COLUMN == commentObjectType)
+    {
```

Comment:
comment max length is 1000, should we make a check to make sure it doesn't go over this limit?  also, can the comment be empty?  Is by adding an empty comment how you drop a comment?

Reply:
Is there a way to get type length of MD table? or I just put a fixed lenght limit here?